src/components/ForgotUserName/ForgotUserNameCtrl.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { withRouter, Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import ForgotUserNameForm from './ForgotUserNameForm';
+import { validateEmail } from '../../helpers';
+
+class ForgotUserNameCtrl extends Component {
+  static propTypes = {
+    mutator: PropTypes.shape({
+      searchUsername: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
+
+  // Expect any MIME type to receive because of the empty body
+  // and no Content-Type header in response
+  static manifest = Object.freeze({
+    searchUsername: {
+      type: 'okapi',
+      path: 'bl-users/forgotten/username',
+      headers: {
+        'accept': '*/*',
+      },
+      fetch: false,
+      throwErrors: false,
+    },
+  });
+
+  state = {
+    isNotValidEmail: false,
+    userDoesNotExist: true,
+    hasErrorsContainer: false,
+    userEmail: '',
+  };
+
+  handleSubmit = values => {
+    this.resetState();
+    const {
+      mutator: { searchUsername },
+    } = this.props;
+    const { userInput } = values;
+    const isValidInput = validateEmail(userInput);
+
+    return isValidInput
+      ? searchUsername
+        .POST({ id: userInput })
+        .then(() => this.handleSuccessfulResponse(userInput))
+        .catch(this.handleBadResponse)
+      : this.setState({ isNotValidEmail: true, hasErrorsContainer: true });
+  };
+
+  handleSuccessfulResponse = (email) => {
+    this.setState({ userDoesNotExist: false, userEmail: email });
+  };
+
+  handleBadResponse = () => {
+    this.setState({ userDoesNotExist: true, hasErrorsContainer: true });
+  };
+
+  resetState = () => {
+    this.setState(
+      {
+        isNotValidEmail: false,
+        userDoesNotExist: true,
+        hasErrorsContainer: false,
+      }
+    );
+  };
+
+  render() {
+    if (this.state.userDoesNotExist === true) {
+      return (
+        <ForgotUserNameForm
+          onSubmit={this.handleSubmit}
+          isNotValid={this.state.isNotValidEmail}
+          hasErrorsContainer={this.state.hasErrorsContainer}
+        />
+      );
+    }
+    return <Redirect to={{
```

Comment:
The component is called` ForgotUserNameCtrl.` I think that in this case by default `ForgotUserNameForm` should be returned. I suggest to do invert if statement and inside of it you should do redirect. Also we may need to rename `userDoesNotExist`. And also I would use state destructuring on the top of the method.

Reply:
Suggest that userDoesNotExist is a logical initial state of the page: we do not know it beforehand